README.md

```diff
@@ -162,132 +169,149 @@ The environment variable `KAFKAJS_LOG_LEVEL` can also be used and it has precede
+```
 
-`label` is a text representation of the log level, example: 'INFO'.
+By default, the producer is configured to distribute the messages with the following logic:
 
-`log` is an object with the following keys: `timestamp`, `logger`, `message`, and the extra keys given by the user. (`logger.info('test', { extra_data: true })`)
+- If a partition is specified in the message, use it
+- If no partition is specified but a key is present choose a partition based on a hash (murmur2) of the key
+- If no partition or key is present choose a partition in a round-robin fashion
+
+The method `send` is used to publish messages to the Kafka cluster.
 
 ```javascript
-{
-  level: 4,
-  label: 'INFO', // NOTHING, ERROR, WARN, INFO, or DEBUG
-  timestamp: '2017-12-29T13:39:54.575Z',
-  logger: 'kafkajs',
-  message: 'Started',
-  // ... any other extra key provided to the log function
-}
-```
+const producer = kafka.producer()
 
-The general structure looks like this:
+async () => {
+  await producer.connect()
+  await producer.send({
+    topic: 'topic-name',
+    messages: [
+      { key: 'key1', value: 'hello world' },
+      { key: 'key2', value: 'hey hey!' }
+    ],
+  })
 
-```javascript
-const MyLogCreator = logLevel => ({ namespace, level, label, log }) => {
-  // Example:
-  // const { timestamp, logger, message, ...others } = log
-  // console.log(`${label} [${namespace}] ${message} ${JSON.stringify(others)}`)
+  // before you exit your app
+  await producer.disconnect()
 }
 ```
 
-Example using [winston](https://github.com/winstonjs/winston):
+Example with a defined partition:
 
 ```javascript
-const { logLevel } = require('kafkajs')
-const winston = require('winston')
-const toWinstonLogLevel = level => switch(level) {
-  case logLevel.ERROR:
-  case logLevel.NOTHING:
-    return 'error'
-  case logLevel.WARN:
-    return 'warn'
-  case logLevel.INFO:
-    return 'info'
-  case logLevel.DEBUG:
-    return 'debug'
-}
-
-const WinstonLogCreator = logLevel => {
-  const logger = winston.createLogger({
-    level: toWinstonLogLevel(logLevel),
-    transports: [
-      new winston.transports.Console(),
-      new winston.transports.File({ filename: 'myapp.log' })
-    ]
+// ...require and connect...
+async () => {
+  await producer.send({
+    topic: 'topic-name',
+    messages: [
+      { key: 'key1', value: 'hello world', partition: 0 },
+      { key: 'key2', value: 'hey hey!', partition: 1 }
+    ],
   })
-
-  return ({ namespace, level, { message, ...extra } }) => {
-    logger.log({
-      level: toWinstonLogLevel(level),
-      message,
-      extra,
-    })
-  }
 }
 ```
 
-Once you have your log creator you can use the `logCreator` option to configure the client:
+The method `send` has the following signature:
 
 ```javascript
-const kafka = new Kafka({
-  clientId: 'my-app',
-  brokers: ['kafka1:9092', 'kafka2:9092'],
-  logLevel: logLevel.ERROR,
-  logCreator: WinstonLogCreator
+await producer.send({
+  topic: <String>,
+  messages: <Array>,
```

Comment:
Maybe this should be `<Message[]>` or `<Map[]<String, String>>` or something instead (I don't know if your syntax is real or made up).